enterprise/internal/campaigns/store/changeset_specs.go

```diff
@@ -414,31 +414,38 @@ type RewirerMapping struct {
 type RewirerMappings []*RewirerMapping
 
 func (rm RewirerMappings) Hydrate(ctx context.Context, store *Store) error {
-	changesetSpecs, _, err := store.ListChangesetSpecs(ctx, ListChangesetSpecsOpts{
-		IDs: rm.ChangesetSpecIDs(),
-	})
-	if err != nil {
-		return err
+	changesetsByID := map[int64]*campaigns.Changeset{}
+	changesetSpecsByID := map[int64]*campaigns.ChangesetSpec{}
+
+	changesetSpecIDs := rm.ChangesetSpecIDs()
+	if len(changesetSpecIDs) > 0 {
+		changesetSpecs, _, err := store.ListChangesetSpecs(ctx, ListChangesetSpecsOpts{
+			IDs: rm.ChangesetSpecIDs(),
```

Comment:
I think we could save regenerating that slice again:

```suggestion
			IDs: changesetSpecIDs,
```

Reply:
whoups 🤦 thanks for being thorough!